test/x509_time_test.c

```diff
@@ -25,6 +25,46 @@ typedef struct {
     int expected;
 } TESTDATA;
 
+/*
+ * Actually, the "loose" mode has been tested in
+ * those time-compare-cases, so we may not test it again.
+ */
+static TESTDATA x509_strict_tests[] = {
+    {
+        "20170217180154Z", V_ASN1_GENERALIZEDTIME,
+        0, 1,
```

Comment:
Hmm, I know what's going wrong: previously I only considered 'checking' the time string format but what is described in the 4.1.2.5. section of RFC 5280 is not implemented in my patch. So the test cases cover only the 'check' part, but no 're-format' the time string.

I need a little more time to add the missing functionality, thanks man!